Content.Client/StationEvents/StationEventManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿#nullable enable
+using Content.Shared.StationEvents;
+using Robust.Shared.Interfaces.Network;
+using Robust.Shared.IoC;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Content.Client.StationEvents
+{
+    class StationEventManager : SharedStationEvent, IStationEventManager
+    {
+        private List<string>? _events;
+        public List<string>? StationEvents
+        {
+            get
+            {
+                if (_events == null)
+                    RequestEvents();
+                return _events;
+            }
+        }
```

Comment:
This field and property are only set to null when disconnecting, but not when restarting the round, and they are not updated when the event list on the server changes. It would be better to remove this entire "null while requesting" pattern and make it request the events on connection, and update them when they update on the server, as they currently can be changed.
If you still need it to be null while requesting, you could change it to an await method instead.

Reply:
I talked with someone about it and the consent was that you should only get the events if you open the admin menu, so don't send it on connect.